lib/plugins/aws/package/compile/events/httpApi/index.test.js

```diff
@@ -461,46 +507,122 @@ describe('HttpApiEvents', () => {
+        expect(cfOutputs).to.not.have.property('HttpApiUrl');
+      });
+      it('Should configure endpoint that attaches to external API', () => {
+        const routeKey = 'POST /some-post';
+        const resource = cfResources[naming.getHttpApiRouteLogicalId(routeKey)];
+        expect(resource.Type).to.equal('AWS::ApiGatewayV2::Route');
+        expect(resource.Properties.RouteKey).to.equal(routeKey);
+        expect(resource.Properties.ApiId).to.equal(apiId);
+      });
+      it('Should configure endpoint integration', () => {
+        const resource = cfResources[naming.getHttpApiIntegrationLogicalId('foo')];
+        expect(resource.Type).to.equal('AWS::ApiGatewayV2::Integration');
+        expect(resource.Properties.IntegrationType).to.equal('AWS_PROXY');
+      });
+      it('Should configure lambda permissions', () => {
+        const resource = cfResources[naming.getLambdaHttpApiPermissionLogicalId('foo')];
+        expect(resource.Type).to.equal('AWS::Lambda::Permission');
+        expect(resource.Properties.Action).to.equal('lambda:InvokeFunction');
+      });
     });
-    it('Should configure lambda permissions', () => {
-      const resource = cfResources[naming.getLambdaHttpApiPermissionLogicalId('foo')];
-      expect(resource.Type).to.equal('AWS::Lambda::Permission');
-      expect(resource.Properties.Action).to.equal('lambda:InvokeFunction');
+
+    describe('disallowed configurations', () => {
+      it('Should not allow defined cors rules', () => {
+        return expect(
+          fixtures
+            .extend('httpApi', {
+              provider: {
+                httpApi: {
+                  id: apiId,
+                  cors: true,
+                },
+              },
+            })
+            .then(fixturePath =>
+              runServerless({
+                cwd: fixturePath,
+                cliArgs: ['package'],
+              })
+            )
+        ).to.eventually.be.rejectedWith(
+          'Cannot setup CORS rules for externally configured HTTP API'
+        );
+      });
+      it('Should not allow defined authorizers', () => {
+        return expect(
+          fixtures
+            .extend('httpApi', {
+              provider: {
+                httpApi: {
+                  id: apiId,
+                  authorizers: {
+                    someAuthorizer: {
+                      identitySource: '$request.header.Authorization',
+                      issuerUrl: 'https://cognito-idp.us-east-1.amazonaws.com/us-east-1_xxx',
+                      audience: 'audiencexxx',
+                    },
+                  },
+                },
+              },
+            })
+            .then(fixturePath =>
+              runServerless({
+                cwd: fixturePath,
+                cliArgs: ['package'],
+              })
+            )
+        ).to.eventually.be.rejectedWith(
+          'Cannot setup authorizers for externally configured HTTP API'
```

Comment:
We should validate against error codes (they’re guaranteed to not change, while message can be improved and that shouldn’t break tests)